pkg/network/setup/podnic.go

```diff
@@ -126,7 +128,17 @@ func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netd
 }
 
 func (l *podNIC) setPodInterfaceCache() error {
-	ifCache := &cache.PodIfaceCacheData{Iface: l.vmiSpecIface}
+	ifCache, err := cache.ReadPodInterfaceCache(l.cacheCreator, string(l.vmi.UID), l.vmiSpecNetwork.Name)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			log.Log.Reason(err).Errorf("failed to read pod interface cache for %s", l.vmiSpecNetwork.Name)
+			return err
+		}
+		log.Log.Reason(err).Warningf("no pod interface cache entry for %s", l.vmiSpecNetwork.Name)
```

Comment:
I don't think the warning is needed here. It is perfectly fine no to have the entry if the cache is not set yet.

Reply:
The cache should have been already set previously due to the state update.
I do not mind removing it though.
Please re-confirm.

Reply:
Done